<!doctype html>
<html>
  <head>
    <meta charset="utf-8">
    <meta http-equiv="X-UA-Compatible" content="chrome=1">
    <title>Allele-frequency-spectrum-simulations by elzaguillot</title>

    <link rel="stylesheet" href="stylesheets/styles.css">
    <link rel="stylesheet" href="stylesheets/pygment_trac.css">
    <meta name="viewport" content="width=device-width, initial-scale=1, user-scalable=no">
    <!--[if lt IE 9]>
    <script src="//html5shiv.googlecode.com/svn/trunk/html5.js"></script>
    <![endif]-->
  </head>
  <body>
    <div class="wrapper">
      <header>
        <h1>Allele-frequency-spectrum-simulations</h1>
        <p>Population genetics; simulations; coalescent</p>

        <p class="view"><a href="https://github.com/elzaguillot/Allele-Frequency-Spectrum-simulations">View the Project on GitHub <small>elzaguillot/Allele-Frequency-Spectrum-simulations</small></a></p>


        <ul>
          <li><a href="https://github.com/elzaguillot/Allele-Frequency-Spectrum-simulations/zipball/master">Download <strong>ZIP File</strong></a></li>
          <li><a href="https://github.com/elzaguillot/Allele-Frequency-Spectrum-simulations/tarball/master">Download <strong>TAR Ball</strong></a></li>
          <li><a href="https://github.com/elzaguillot/Allele-Frequency-Spectrum-simulations">View On <strong>GitHub</strong></a></li>
        </ul>
      </header>
      <section>
        <h1>
<a id="allele-frequency-spectrum-simulations" class="anchor" href="#allele-frequency-spectrum-simulations" aria-hidden="true"><span class="octicon octicon-link"></span></a>Allele Frequency Spectrum Simulations</h1>

<h3>
<a id="what-is-it" class="anchor" href="#what-is-it" aria-hidden="true"><span class="octicon octicon-link"></span></a>What is it?</h3>

<p>This repository contains the code to simulate the allele ('haplotype') frequency spectrum (Figure 1) observed in Central Asia by Balaresque and colleagues ("Y chromosome descent clusters and male differential reproductive success: Young lineage expansions dominate Asian pastoral populations").</p>

<p>The aim is to check whether the observed distribution varies from neutral expectations.</p>

<h3>
<a id="how-to-run-the-codes" class="anchor" href="#how-to-run-the-codes" aria-hidden="true"><span class="octicon octicon-link"></span></a>How to run the codes?</h3>

<p>Part 1: Finding theta</p>

<p>The aim is to determine which value of theta (the scaled population mutation rate) matches the observed allelic diversity.  The code for this analysis is provided in the 'theta' repository.</p>

<p>Part 2: Finding the variance of the allele frequency distribution</p>

<p>The aim is to place confidence intervals on the observed allele frequency distribution from one million coalescent datasets.  The code for this analysis is provided in the 'AFS' repository.</p>

<p>Note: runtimes for both parts of this analysis are long (hours to days).  Y chromosome coalescent trees are generated with ms and transformed to 8 fully-linked STR loci using microsat.  Output files are parsed with Python and R.  For each section, the entire analysis has been automated as a bash script.</p>

<h3>
<a id="requirements" class="anchor" href="#requirements" aria-hidden="true"><span class="octicon octicon-link"></span></a>Requirements</h3>

<p>The following computer requirements to needed to run the code:</p>

<ul>
<li>UNIX operating system</li>
<li>R</li>
<li>Python</li>
<li>ms: <a href="http://home.uchicago.edu/%7Erhudson1/source/mksamples.html">http://home.uchicago.edu/~rhudson1/source/mksamples.html</a>
</li>
<li>microsat: <a href="http://massey.genomicus.com/software.html#microsat">http://massey.genomicus.com/software.html#microsat</a>
</li>
</ul>

<h3>
<a id="references" class="anchor" href="#references" aria-hidden="true"><span class="octicon octicon-link"></span></a>References</h3>

<p>Hudson R. 2002. Generating samples under a Wrightâ€“Fisher neutral model of genetic variation. Bioinformatics 18:337-338.</p>

<p>Balaresque P, et al. 2015. Y chromosome descent clusters and male differential reproductive success: Young lineage expansions dominate Asian pastoral populations. European Journal of Human Genetics, In Press.</p>

<h3>
<a id="authors-and-contributors" class="anchor" href="#authors-and-contributors" aria-hidden="true"><span class="octicon octicon-link"></span></a>Authors and Contributors</h3>

<p>Created by <a href="https://github.com/elzaguillot" class="user-mention">@elzaguillot</a> and <a href="https://github.com/mpcox" class="user-mention">@mpcox</a></p>
      </section>
      <footer>
        <p>This project is maintained by <a href="https://github.com/elzaguillot">elzaguillot</a></p>
        <p><small>Hosted on GitHub Pages &mdash; Theme by <a href="https://github.com/orderedlist">orderedlist</a></small></p>
      </footer>
    </div>
    <script src="javascripts/scale.fix.js"></script>
    
  </body>
</html>